components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/util/LightFileImportComponent.java

```diff
@@ -0,0 +1,1135 @@
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #hasComponents()
+     */
+    @Override
+    public boolean hasComponents() {
+        return components != null && components.size() > 0;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #getImportStatus()
+     */
+    @Override
+    public ImportStatus getImportStatus() {
+        if (getFile().isFile()) {
+            if (hasImportFailed())
+                return ImportStatus.FAILURE;
+            return resultIndex;
+        }
+        if (components == null || components.size() == 0) {
+            if (image instanceof Boolean) {
+                if (getFile().isDirectory()) {
+                    return ImportStatus.SUCCESS;
+                } else {
+                    if (!status.isMarkedAsCancel()
+                            && !status.isMarkedAsDuplicate())
+                        return ImportStatus.FAILURE;
+                }
+            }
+            return resultIndex;
+        }
+
+        Collection<LightFileImportComponent> values = components.values();
+        int n = components.size();
+        int count = 0;
+        synchronized (components) {
+            Iterator<LightFileImportComponent> i = values.iterator();
+            while (i.hasNext()) {
+                if (i.next().hasImportFailed())
+                    count++;
+            }
+        }
+        if (count == n)
+            return ImportStatus.FAILURE;
+        if (count > 0)
+            return ImportStatus.PARTIAL;
+        return ImportStatus.SUCCESS;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #hasToRefreshTree()
+     */
+    @Override
+    public boolean hasToRefreshTree() {
+        if (getFile().isFile()) {
+            if (hasImportFailed())
+                return false;
+            switch (type) {
+            case PROJECT:
+            case NA:
+                return true;
+            default:
+                return false;
+            }
+        }
+        if (components == null)
+            return false;
+        if (importable.isFolderAsContainer() && type != ContainerType.PROJECT) {
+            Collection<LightFileImportComponent> values = components.values();
+            synchronized (components) {
+                Iterator<LightFileImportComponent> i = values.iterator();
+                while (i.hasNext()) {
+                    if (i.next().toRefresh())
+                        return true;
+                }
+            }
+            return false;
+        }
+        return true;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #toRefresh()
+     */
+    @Override
+    public boolean toRefresh() {
+        /*
```

Comment:
Guessing this is work in progress.

Reply:
If so it has been work in progress since a long time, as far as I can see this always has been commented out.